pkg/controller/storagecluster/noobaa_system_reconciler.go

```diff
@@ -125,3 +129,70 @@ func (r *ReconcileStorageCluster) newNooBaaSystem(sc *ocsv1.StorageCluster, reqL
 
 	return nb
 }
+
+// Delete all nooba systems in the namespace
+func (r *ReconcileStorageCluster) deleteNoobaaSystemsAndWait(sc *ocsv1.StorageCluster) error {
+	noobaas := &nbv1.NooBaaList{}
+	listOptions := &client.ListOptions {
+		Namespace: sc.Namespace,
+	}
+	err := r.client.List(context.TODO(), listOptions, noobaas)
+	if err != nil {
+		return err
+	}
+
+	for _, noobaa := range noobaas.Items {
+		err = r.client.Delete(context.TODO(), &noobaa)
+		if err != nil && !errors.IsNotFound(err) {
+			return err
+		}
+	}
+
+	labelSelector, err := labels.Parse("noobaa-core in (noobaa)")
+	if err != nil {
+		return err
+	}
+
+	pvcs, err := r.k8sClient.CoreV1().PersistentVolumeClaims(sc.Namespace).List(metav1.ListOptions{
+		LabelSelector: labelSelector.String(),
+	})
+	if err != nil {
+		return err
+	}
+
+	for _, pvc := range pvcs.Items {
+		err := r.k8sClient.CoreV1().PersistentVolumeClaims(sc.Namespace).Delete(pvc.Name, &metav1.DeleteOptions{})
```

Comment:
we don't own these PVCs.  The noobaa object should be deleting these, not us.

Reply:
in case it wasn't clear, it's fine to wait on these pvcs to disappear if we need to though. we just shouldn't be deleting them.

Reply:
These PVCs are owned by Noobaa and we do need to wait until they disappear.